openeo/local/collections.py

```diff
@@ -0,0 +1,250 @@
+    y_min = data[y_dim].min().item(0)
+    y_max = data[y_dim].max().item(0)
+
+    crs_present = False
+    coords = list(data.coords)
+    if 'spatial_ref' in coords:
+        # bands.remove('crs')
+        crs_present = True
+    # TODO: list bands if more available
+    bands = []
+    if 'band' in coords:
+        bands = list(data['band'].values)
+        if len(bands)>0:
+            # The JSON decoder does not handle npint types, we need to convert them in advance
+            if isinstance(bands[0],np.int8) or \
+                isinstance(bands[0],np.int16) or \
+                isinstance(bands[0],np.int32) or \
+                isinstance(bands[0],np.int64):
+                bands = [int(b) for b in bands]
+    extent = {}
+    if crs_present:
+        if 'crs_wkt' in data.spatial_ref.attrs:
+            transformer = Transformer.from_crs(data.spatial_ref.attrs['crs_wkt'], 'epsg:4326')
+            lat_min,lon_min = transformer.transform(x_min,y_min)
+            lat_max,lon_max = transformer.transform(x_max,y_max)
+            extent['spatial'] = {'bbox': [[lon_min, lat_min, lon_max, lat_max]]}
+
+    if t_dim is not None:
+        t_min = str(data[t_dim].min().values)
+        t_max = str(data[t_dim].max().values)
+        extent['temporal'] = {'interval': [[t_min,t_max]]}
+
+    metadata['extent'] = extent
+
+    t_dimension = {}
+    if t_dim is not None:
+        t_dimension = {t_dim: {'type': 'temporal', 'extent':[t_min,t_max]}}
+
+    x_dimension = {x_dim: {'type': 'spatial','axis':'x','extent':[x_min,x_max]}}
+    y_dimension = {y_dim: {'type': 'spatial','axis':'y','extent':[y_min,y_max]}}
+    if crs_present:
+        if 'crs_wkt' in data.spatial_ref.attrs:
+            x_dimension[x_dim]['reference_system'] = data.spatial_ref.attrs['crs_wkt']
+            y_dimension[y_dim]['reference_system'] = data.spatial_ref.attrs['crs_wkt']
+
+    b_dimension = {}
+    if len(bands)>0:
+        b_dimension = {'bands': {'type': 'bands', 'values':bands}}
+
+    metadata['cube:dimensions'] = {**t_dimension,**x_dimension,**y_dimension,**b_dimension}
+
+    return metadata
+
+def _get_netcdf_zarr_collections(local_collections_path):
+    if isinstance(local_collections_path,str):
+        local_collections_path = [local_collections_path]
+    local_collections_list = []
+    for flds in local_collections_path:
+        local_collections_netcdf_zarr = [p for p in Path(flds).rglob('*') if p.suffix in  ['.nc','.zarr']]
+        for local_file in local_collections_netcdf_zarr:
+            try:
+                metadata = _get_netcdf_zarr_metadata(local_file)
+                local_collections_list.append(metadata)
+            except Exception as e:
+                _log.error(e)
+                continue
+    
+    local_collections_dict = {'collections':local_collections_list}
+    
+    return local_collections_dict
+
+def _get_geotiff_collections(local_collections_path):
+    if isinstance(local_collections_path,str):
```

Comment:
again, this looks very similar to `_get_netcdf_zarr_collections`
I guess this can be generalized

Reply:
Merged the two functions into one